engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/flink/test/MiniClusterExecutionEnvironment.scala

```diff
@@ -87,6 +89,25 @@ class MiniClusterExecutionEnvironment(
     waitForJobStateWithAdditionalCheck(jobID, name, {}, expectedState: _*)(patience)
   }
 
+  def waitForJobStatusWithAdditionalCheck(
```

Comment:
For batch jobs even after the job is initialized, not all vertices are available at all times. There are moments when source is FINISHED and next vertices are not even registered leading to a passing check. Thats why for FINISHED jobs a better check is to check `executionGraph.getState`